pilot/pkg/proxy/envoy/proxy.go

```diff
@@ -140,9 +143,18 @@ func (e *envoy) Run(config interface{}, epoch int, abort <-chan error) error {
 	if err := cmd.Start(); err != nil {
 		return err
 	}
+	// record envoy process number
+	e.processNum.Inc()
 	done := make(chan error, 1)
 	go func() {
-		done <- cmd.Wait()
+		err := cmd.Wait()
+		e.processNum.Dec()
+		if err == nil {
+			if e.processNum.Load() == 0 {
+				err = fmt.Errorf("envoy process is killed, maybe by SIGTERM")
```

Comment:
This error message should make clear the envoy process was not killed by pilot-agent.